### ISSUE TITLE
SC2154 fires on associative array indices.

### ISSUE THREAD

**Issue by deryni** [koalaman/shellcheck#359]
When run on this script

    #!/bin/bash

    declare -A arr=()

    declare -p arr
    arr[k1]=lfkej
    declare -p arr

    # Silence SC2034 warning about arr being unused.
    : "${arr[@]}"

shellcheck reports

    In arr.sh line 6:
    arr[k1]=lfkej
        ^-- SC2154: k1 is referenced but not assigned.

but `arr` is an associative array and the contents of `[]` for them are not arithmetically evaluated.

**Comment by pbiggar** [comment]
I can verify this. I worked around it by declaring inline:

```declare -A array=( [k1]=lfkej )```

**Comment by cyril-bouthors** [comment]
Nice work-around @pbiggar, I confirm it works, thanks!

**Comment by strizhechenko** [comment]
I have same problem
2 @pbiggar - inline declare breaks readability in big arrays(

**Comment by Ghoughpteighbteau** [comment]
@strizhechenko 

but you can circumvent that like this:
```
#color matching criteria (case insensitive)
declare -A colorm=(
	[red]="red"
	[orange]="orange"
	[yellow]="yellow"
	[green]="green"
	[blue]="blue"
	[black]="black"
	[violet]="violet\|purple\|magenta\|pink"
	[all]="all of \(the above\|them\)"
)
```

**Comment by koalaman** [comment]
Fixed in 07fd5724

**Comment by pbiggar** [comment]
Awesome, thanks!
